tests/behavior/transforms/remap.toml

```diff
@@ -46,6 +46,44 @@
       "b[1].equals" = "<null>"
       "b[2].equals" = "two"
 
+[transforms.push_to_array]
+  inputs = []
+  type = "remap"
+  source = """
+    .fruits = push(.fruits, .fruit)
+  """
+[[tests]]
+  name = "push_to_array"
+  [tests.input]
+    insert_at = "push_to_array"
+    type = "log"
+    [tests.input.log_fields]
+      fruits = ["apple", "orange", "banana"]
+      fruit = "mango"
+  [[tests.outputs]]
+    extract_from = "push_to_array"
+    [[tests.outputs.conditions]]
+      "fruits.equals" = ["apple", "orange", "banana", "mango"]
+
+[transforms.append_to_array]
+  inputs = []
+  type = "remap"
+  source = """
+    .fruits = append(.fruits1, .fruits2)
+  """
+[[tests]]
+  name = "append_to_array"
+  [tests.input]
+    insert_at = "append_to_array"
+    type = "log"
+    [tests.input.log_fields]
+      fruits1 = ["apple", "orange"]
+      fruits2 = ["banana", "mango"]
+  [[tests.outputs]]
+    extract_from = "append_to_array"
+    [[tests.outputs.conditions]]
+      "fruits.equals" = ["apple", "orange", "banana", "mango"]
+
```

Comment:
nit: to reduce merge conflicts, we prefer adding new tests to the bottom of the file. Also, we name function tests `remap_function_*` by convention, which makes it easier to know how to run a single function test.